lib/api/controllers/securityController.js

```diff
@@ -753,3 +912,145 @@ function formatUserSearchResult(kuzzle, formatter, raw) {
       return users;
     });
 }
+
+
+/**
+ * Creates a new User object in Kuzzle's database layer
+ *
+ * @param {Kuzzle} kuzzle
+ * @param {Request} request
+ * @param {boolean} restricted
+ * @returns {Promise<Object>}
+ */
+function createUser (kuzzle, request, restricted) {
+  assertHasBody(request);
+  assertIdStartsNotUnderscore(request);
+  assertBodyHasAttribute(request, 'content');
+  assertBodyAttributeType(request, 'content', 'object');
+
+  if (request.input.body.credentials) {
+    assertBodyAttributeType(request, 'credentials', 'object');
+  }
+
+  if (restricted) {
+    if (request.input.body.content.profileIds) {
+      throw new BadRequestError('The request must not specify the body attribute "content.profileIds".');
+    }
+  }
+  else {
+    if (!request.input.body.content.profileIds) {
+      throw new BadRequestError('The request must specify a body attribute "content.profileIds".');
+    }
+    if (!Array.isArray(request.input.body.content.profileIds)) {
+      throw new BadRequestError('The request must specify the body attribute "content.profileIds" of type "array".');
+    }
+  }
+
+  const pojoUser = request.input.body.content;
+  pojoUser._id = request.input.resource._id || uuid.v4();
+
+  if (restricted) {
+    pojoUser.profileIds = kuzzle.config.security.restrictedProfileIds;
+  }
+
+  return Bluebird.coroutine(persistUserAndCredentials)(kuzzle, request, pojoUser);
+}
+
+/**
+ * @param {Kuzzle} kuzzle
+ * @param {Request} request
+ * @param {object} pojoUser
+ */
+function* persistUserAndCredentials (kuzzle, request, pojoUser) {
+  const
+    user = new User(),
+    loadedUser = yield Bluebird.resolve(kuzzle.repositories.user.load(pojoUser._id));
+  let
+    strategies = [],
+    errorStep = null,
+    creationError = null,
+    errorMessage,
+    createdUser,
+    errors;
+
+  if (loadedUser !== null && loadedUser._id !== kuzzle.repositories.user.anonymous()._id) {
+    throw new BadRequestError(`user "${pojoUser._id}" already exist.`);
```

Comment:
a `PreconditionError` (introduced in [kuzzle-common-objects 3.0.3](https://github.com/kuzzleio/kuzzle-common-objects/releases/tag/3.0.3)) is here more pertinent than a `BadRequestError`.